backend/src/main/java/quarano/actions/DiaryEventListener.java

```diff
@@ -23,14 +26,20 @@
 	private final @NonNull AnomaliesProperties config;
 	private final @NonNull ActionItemRepository items;
 	private final @NonNull DiaryManagement diaryManagement;
+	private final @NonNull TrackedCaseRepository cases;
+
 
 	@EventListener
 	void on(DiaryEntryAdded event) {
 
 		var entry = event.getEntry();
 
-		handleDiaryEntryForBodyTemprature(entry);
-		handleDiaryEntryForCharacteristicSymptoms(entry);
+		boolean isContactCase = cases.findByTrackedPerson(entry.getTrackedPersonId())
```

Comment:
I think we should use a dedicated repository query method for that here to avoid the entire `TrackedCase` to be materialized just to evaluate a single boolean flag on it.

Reply:
I partially agree here. Partially because the system is built this visibly very low precentage of bare SQL queries. My impression is that quarano attempts to go as "SQL-less" as possible. This is what you find otherwise in code:

return encounters.stream().map(Encounter::getContact).distinct().count();

So what is the common agremment? Do we want dedicated queries for "small things"?